graylog2-web-interface/src/views/components/visualizations/bar/BarVisualization.tsx

```diff
@@ -93,27 +93,60 @@ const BarVisualization = makeVisualization(({
       .join(humanSeparator),
     ), [mapKeys, rowPivotFields]);
 
-  const _seriesGenerator: Generator = useCallback(({ type, name, labels, values, originalName }): ChartDefinition => ({
-    type,
-    name,
-    x: _mapKeys(labels),
-    y: values,
-    opacity,
-    originalName,
-  }), [_mapKeys, opacity]);
+  const { getExtendedBarsGeneratorSettings } = useExtendedChartGeneratorSettings({ config, effectiveTimerange, barmode });
+
+  const _seriesGenerator: Generator = useCallback(({ type, name, labels, values, originalName, total, idx }): ChartDefinition => {
+    const opacity = visualizationConfig?.opacity ?? 1.0;
+    const mappedKeys = _mapKeys(labels);
+
+    const getData = () => ({
```

Comment:
The function is created and called immediately, unnecessarily. I think we should revert this to just returning the object.

Reply:
There was an issue when it was an object, in some cases updating didn't change the layout. Looks like there was some cashing or something. But I checked it and it seems like it's no longer an issue.